util/timer.h

```diff
@@ -56,28 +56,27 @@ class Timer {
   // the function takes 1000us to run. If it starts at time [now]us, then it
   // finishes at [now]+1000us, 2nd run starting time will be at [now]+3000us.
   // repeat_every_us == 0 means do not repeat.
-  void Add(std::function<void()> fn,
-           const std::string& fn_name,
-           uint64_t start_after_us,
-           uint64_t repeat_every_us) {
+  bool Add(std::function<void()> fn, const std::string& fn_name,
+           uint64_t start_after_us, uint64_t repeat_every_us) {
+    InstrumentedMutexLock l(&mutex_);
```

Comment:
Nit: we can avoid dynamic memory allocation inside this mutex by calling NowMicros() and setting next run time after mutex acquision.
```
std::unique_ptr<> fn_info(new FunctionInfo(..., dummy_time,...));
InstrumentedMutexLock l(&mutex_);
now = NowMicros();
fn_info->SetNextRunTime(now + start_after_us);
...
```